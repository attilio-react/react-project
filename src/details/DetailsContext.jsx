import React from 'react';

export const DetailsContext = React.createContext(
   {
     movie: {},
     relatedMovies: []
   }
)


