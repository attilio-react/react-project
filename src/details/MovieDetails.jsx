import React from 'react';

import {DetailsContext} from './DetailsContext.jsx'

import {Button} from 'Common/Button.jsx'

import {MovieData} from './MovieData.jsx'
import {RelatedMovies} from './RelatedMovies.jsx'

class MovieDetails extends React.PureComponent {
  render () {
    return (<DetailsContext.Consumer>
            {ctx => (
                <>
                &nbsp;
                &nbsp;
                <Button caption='Search' onClick={ctx.searchClickCb} />
                <MovieData />
                <RelatedMovies />
                </>
            )}
            </DetailsContext.Consumer>)
  }

}


export {MovieDetails}


