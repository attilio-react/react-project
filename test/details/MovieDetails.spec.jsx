import * as React from 'react';
import {mount} from 'enzyme';
import renderer from 'react-test-renderer';


import {MovieDetails} from '|details/MovieDetails.jsx';
import {DetailsContext} from '|details/DetailsContext.jsx';


const CONTEXT = {
    movie: {
        poster_path: 'http://hello.com/poster3.png',
        title: 'Movie Title3',
        vote_average: 4.3,
        release_date: '2013-01-01',
        runtime: 129,
        overview: 'This is a third movie about ....'
    },
    relatedGenres: ['Adventure', 'War', 'Thriller'],
    relatedMovies: [ 
        {
            poster_path: 'http://hello.com/poster.png',
            title: 'Movie Title',
            vote_average: 4.4,
            release_date: '2011-01-01',
            runtime: 129,
            overview: 'This is a movie about ....',
            genres: ['Horror', 'Thriller']
        },
        {
            poster_path: 'http://hello.com/poster2.png',
            title: 'Another Movie Title',
            vote_average: 4.4,
            release_date: '2012-02-02',
            runtime: 139,
            overview: 'This is another movie about ....',
            genres: ['War', 'History']
        }
    ]
}


describe('MovieDetails rendering', () => {
  it('Should render', () => {
    const 
        component = renderer.create(
            <DetailsContext.Provider value = {CONTEXT}>
                <MovieDetails />
            </DetailsContext.Provider>
        ),
        json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});


