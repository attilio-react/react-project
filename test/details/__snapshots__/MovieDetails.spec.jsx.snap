// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails rendering Should render 1`] = `
Array [
  "   ",
  <input
    style={
      Object {
        "textDecoration": "none",
      }
    }
    type="button"
    value="Search"
  />,
  <div>
    <img
      src="http://hello.com/poster3.png"
    />
    <p>
      <span>
        Movie Title3
      </span>
    </p>
    <p>
      <span>
        4.3
      </span>
    </p>
    <p>
      <span>
        2013-01-01
      </span>
    </p>
    <p>
      <span>
        129
      </span>
    </p>
    <p>
      <span>
        This is a third movie about ....
      </span>
    </p>
  </div>,
  <div>
    <span>
      Films by genre Adventure, War, Thriller
    </span>
    <div
      style={
        Object {
          "border": "1px solid black",
        }
      }
    >
      <a
        onClick={[Function]}
      >
        <p>
          <img
            src="http://hello.com/poster.png"
          />
        </p>
        <p>
          <span>
            2011-01-01
          </span>
           
          <span>
            Horror, Thriller
          </span>
        </p>
        <p>
          <span>
            Movie Title
          </span>
        </p>
      </a>
    </div>
    <div
      style={
        Object {
          "border": "1px solid black",
        }
      }
    >
      <a
        onClick={[Function]}
      >
        <p>
          <img
            src="http://hello.com/poster2.png"
          />
        </p>
        <p>
          <span>
            2012-02-02
          </span>
           
          <span>
            War, History
          </span>
        </p>
        <p>
          <span>
            Another Movie Title
          </span>
        </p>
      </a>
    </div>
  </div>,
]
`;
